app/controllers/report_controller/saved_reports.rb

```diff
@@ -157,10 +157,10 @@ def set_saved_reports_condition(rep_id = nil)
 
     # Admin users can see all saved reports
     unless admin_user?
-      cond[0] << " AND miq_group_id=?"
-      cond.push(current_group_id)
+      cond[0] << " AND miq_group_id IN (?)"
+      cond.push(current_user.miq_groups.collect(&:id))
     end
 
-    cond.flatten
```

Comment:
I think you still want to flatten the array before returning it here. When the condition is called with a nested array you'll get a deprecation warning roo:

DEPRECATION WARNING: Passing a nested array to Active Record finder methods is deprecated and will be removed. Flatten your array before using it for 'IN' conditions.